src/LogLevelConfiguration.php

```diff
@@ -65,6 +65,11 @@ private function determineApplicableLogLevel(float $durationInSeconds) : ?string
         //Acquire a common / non-associative array with all the thresholds
         $durationThresholds = array_values($this->logLevelMapping);
 
+        $exactDurationMatch = array_search($durationInMilliseconds, $this->logLevelMapping);
```

Comment:
1. this can/should be done "as early as possible" in the function?
1. What is the chance that a duration (float) is exactly matched in a real-world situation? I think that chance is near 0... So what justifies this check? Being able to test it?

Reply:
> 2. What is the chance that a duration (float) is exactly matched in a real-world situation? I think that chance is near 0... So what justifies this check? Being able to test it?

i agree that it is near 0. i wouldn't say that the justification is testability, rather accuracy. i tested an edge and the value i expected to get was the not the actual value.

> 1. this can/should be done "as early as possible" in the function?

i think you're correct in assuming the chances are near 0. would this be more palatable at the very end of the function? before null is returned?

Reply:
You mention two very valid points 😉. Indeed, moving the "exact match" scenario to the end seems a good place for such edge case 👍